### PR TITLE
#174797528 Manage available requests

### PR DESCRIPTION
**Pull request by ericus123**
**What does this PR do?**

Display requests created by manager's direct report so that he can make a decision

**Description of Task to be completed?**
  - create a private route to display requests directly sent to a manager.
  - create a private route for a manager to make a decision on  requests directly sent to him.
  - validate decision route.
 - document the feature using swagger 
 - test the feature

**How should this be manually tested?**

  - clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
  - switch your working directory `cd script-heroes-bn-backend`
  - checkout the branch `git checkout ft-manage-requests-for-approval-174797528`
  - create admin role by running `npm run role:create` and then run `npm run admin:create` 
  - create manager role:
      - login as an admin and use endpoint  `http://localhost:3000/api/role/register`
     

>   request body

```
  {
      "name":"MANAGER",
      "description":"Manager of Barefoot Nomad"
 }
```
 - assign manager role to a user  `http://localhost:3000/api/role/assign`
  

>  request body

  ` {
      "userRole":"MANAGER",
      "email":"user@email.com"
  }`
  - get requests as a manager` http://localhost:3000/api/requests/`
  - make a decision on a request as a manager `http://localhost:3000/api/requests/:id/:decision`

> decision can either be approve or decline

**What are the relevant pivotal tracker stories?**

   [#174797528](https://www.pivotaltracker.com/story/show/174797528)

**Screenshots (if appropriate)**
Screenshot from 2020-10-27 21-42-29
Screenshot from 2020-10-27 21-41-51

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -217,5 +217,14 @@
...
+  "Accommodation deleted successfully": "Accommodation deleted successfully",
+	"Check-out date is required": "CHeck-out date is required",
+	"Unauthorized request": "Unauthorized request",
+	"Request approved succesfully": "Request approved successfully",
+	"Request declined succesfully": "Request declined successfully",
+	"Request doesn't exist": "Request doesn't exist",
+	"Decision must be one of approve or decline": "Decision must be one of approve or decline",
+	"Request declined successfully": "Request declined successfully",
+	"Request approved successfully": "Request approved successfully"
 }
```
Comment (Habinezajanvier) [inline locales/en.json]: Please use the normal identation

File: test/controllers/manageRequests.test.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        firstName: 'First',
+        lastName: 'Last',
+        email: 'manager2@gmail.com',
+        phoneNumber: '3444444444',
+        password: hashedPassword,
+        isVerified: true,
+        roleId: role.id
+      };
+      await User.create(manager1);
+      await User.create(manager2);
```
Comment (Habinezajanvier) [inline test/controllers/manageRequests.test.js]: Couldn't it be possible to seed these managers and use them during testing

File: src/middlewares/auth.middleware.js [hunk 3]
```diff
@@ -49,6 +50,18 @@ class AuthMiddleware {
...
+      const userParam = await decode(token);
+  
+      const { roleId } = userParam;
+      if (roleId === null) return res.status(401).json({ msg: res.__('Unauthorized request') });
+  
+      const role = await RoleService.findRoleById(roleId);
+  
+      if (role.name === 'MANAGER') return next();
+      return res.status(401).json({ msghaha: res.__('Unauthorized request') });
+    }
```
Comment (jpirumvaa) [inline src/middlewares/auth.middleware.js]: Please check well this typo, `msghaha`

**Review by jpirumvaa — CHANGES_REQUESTED** [review]

File: src/middlewares/auth.middleware.js (hunk 3 above)
Comment (ericus123) [reply src/middlewares/auth.middleware.js]: Yes,done now

**Review by elvisiraguha — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Smells good to me

File: src/middlewares/auth.middleware.js (hunk 3 above)
Comment (jpirumvaa) [reply src/middlewares/auth.middleware.js]: Well done

**Review by jpirumvaa — APPROVED** [review]

**Review by valens10 — APPROVED** [review]

**Pull request merged**